app/src/main/java/lvc/com/movies/util/AppExecutors.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package lvc.com.movies.util;
+
+import android.app.Application;
+import android.os.Handler;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+/**
+ * Global executor pools for the whole application.
+ * <p>
+ * Grouping tasks like this avoids the effects of task starvation (e.g. disk reads don't wait behind
+ * webservice requests).
+ */
+
+public class AppExecutors {
```

Comment:
Why do we need this class for this project? Why is it copied into the project?

Reply:
AppExecutors provides Executors for different moments in the app.
diskIO is an Executor to execute Runnables related to database.
networkIO is an Executor to execute Runnables related to http requests and it has a thread Pool.
mainThread is an Executor to execute Runnables related to UI. 
It works very well in this project because I had to get data from different places (Web and local database) and after I merged this data it will be presented in the UI.
I think I should have created a Library module to keep all classes that could be portable for other projects there, and this class is one of them.